src/server/worker/datum/create_stream.go

```diff
@@ -0,0 +1,93 @@
+package datum
+
+import (
+	"context"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/storage/renew"
+	"github.com/pachyderm/pachyderm/v2/src/internal/task"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"github.com/pachyderm/pachyderm/v2/src/pps"
+)
+
+const (
+	// The target number of files per shard. A smaller value returns datums to
+	// the client faster. A larger value reduces the overhead of creating datums.
+	ShardNumFiles = 10000
+)
+
+type createDatumStream struct {
+	ctx               context.Context
```

Comment:
Don't embed a context.Context in a struct: https://go.dev/blog/context-and-structs

Reply:
I can certainly change the function calls to pass contexts there. The reason why I embedded the context in the struct is because I need `ctx` in the `Iterate` function in `iterator.go`. `Iterate` is part of an interface, so rather than changing the function signature, I thought I'd be consistent with what the other `Iterate` functions are doing (see [here](https://github.com/pachyderm/pachyderm/blob/master/src/server/worker/datum/iterator.go#L91) and [here](https://github.com/pachyderm/pachyderm/blob/master/src/server/worker/datum/iterator.go#L181)).

Reply:
I think the createDatumStreamIterator is what should embed the context.